src/seedcase_sprout/core/extract_resource_properties.py

```diff
@@ -0,0 +1,53 @@
+from pathlib import Path
+
+from frictionless import describe
+
+from seedcase_sprout.core.check_is_file import check_is_file
+from seedcase_sprout.core.check_is_supported_format import check_is_supported_format
+from seedcase_sprout.core.checks import check_resource_properties
+from seedcase_sprout.core.properties import ResourceProperties
+
+
+def extract_resource_properties(data_path: Path) -> ResourceProperties:
+    """Extracts resource properties from a raw data file.
+
+    This function takes the data file found at the `data_path` location and
+    extracts properties from the file into a `ResourceProperties` object. This
+    function is often followed by `edit_resource_properties()` to fill in any
+    remaining missing fields, like the `path` property field.  Usually, you use
+    either this function or the `create_resource_properties()` function to
+    create the initial resource properties for a specific (new) data resource.
+
+    Args:
+        data_path: The path to a raw data file of a supported format.
+
+    Returns:
+        Outputs a `ResourceProperties` object. Use `write_resource_properties()`
+            to save the object to the `datapackage.json` file.
+    """
+    check_is_file(data_path)
+    check_is_supported_format(data_path)
+
+    properties = describe(data_path).to_dict()
+
+    if properties["format"] == "tsv":
+        _clean_tsv_properties(properties)
```

Comment:
```suggestion
    properties.pop("dialect", None)
```
Just to explain: we recently [decided](https://github.com/seedcase-project/seedcase-sprout/issues/1032#issuecomment-2668266424) with Luke to remove the `dialect` property, so it makes sense to delete it here.

But if frictionless could determine that the file has another dialect than the default, we could alert the user?

Reply:
If we find that it would be nice to alert the user, we can add it later on 👍